UnitsNet/GeneratedCode/Quantities/Angle.g.cs

```diff
@@ -865,11 +872,31 @@ public double As(UnitSystem unitSystem)
         /// <summary>
         ///     Converts this Angle to another Angle with the unit representation <paramref name="unit" />.
         /// </summary>
+        /// <param name="unit">The unit to convert to.</param>
         /// <returns>A Angle with the specified unit.</returns>
         public Angle ToUnit(AngleUnit unit)
         {
-            var convertedValue = GetValueAs(unit);
-            return new Angle(convertedValue, unit);
+            return ToUnit(unit, DefaultConversionFunctions);
+        }
+
+        /// <summary>
+        ///     Converts this Angle to another Angle using the given <paramref name="unitConverter"/> with the unit representation <paramref name="unit" />.
+        /// </summary>
+        /// <param name="unit">The unit to convert to.</param>
+        /// <param name="unitConverter">The <see cref="UnitConverter"/> to use for the conversion.</param>
+        /// <returns>A Angle with the specified unit.</returns>
+        public Angle ToUnit(AngleUnit unit, UnitConverter unitConverter)
+        {
+            if(Unit == unit)
+                return this;
+
+            var inBaseUnits = ToUnit(BaseUnit);
+
+            if(!unitConverter.TryGetConversionFunction<Angle>(inBaseUnits.Unit, unit, out var conversionFunction))
```

Comment:
I like it. 

As a very minor performance improvement, can we call the innermost method overload? I think it has to go through 2 additional `TryGetConversionFunction` methods to get to the most specific one.

Reply:
Besides the minor detail of failing tests, I think this looks good?

So in order to extend this, we can simply call `Acceleration.DefaultConversionFunctions.SetXXX()` on startup?

Reply:
> Need to formalize this more as we should really use QuantityInfo for the parameters.

What did you mean by this?

Reply:
> I like it.
> 
> As a very minor performance improvement, can we call the innermost method overload? I think it has to go through 2 additional `TryGetConversionFunction` methods to get to the most specific one.

Done!

Reply:
> Besides the minor detail of failing tests, I think this looks good?
> 
> So in order to extend this, we can simply call `Acceleration.DefaultConversionFunctions.SetXXX()` on startup?
> 
> Will there be a racing condition with the initialization of the global static `UnitConverter.Default`?

You certainly could. I do worry a bit though about that causing unintended consequences, especially since ```GetValueAs``` uses the ```DefaultConversionFunctions```. This would affect:
- operator +
- operator -
- operator <=
- operator >=
- operator <
- operator >
- CompareTo
- Equals
- As

That said, if you write a custom conversion function, arguably it **should** end up being used for * for example.